packages/calcite-components/src/components/panel/panel.tsx

```diff
@@ -288,16 +298,36 @@ export class Panel
 
   panelKeyDownHandler = (event: KeyboardEvent): void => {
     if (this.closable && event.key === "Escape" && !event.defaultPrevented) {
-      this.close();
+      this.closed = true;
       event.preventDefault();
     }
   };
 
-  close = (): void => {
+  private handleCloseClick = (): void => {
     this.closed = true;
     this.calcitePanelClose.emit();
   };
 
+  open = (): void => {
+    this.isClosed = false;
+  };
+
+  close = async (): Promise<void> => {
+    const beforeClose = this.beforeClose ?? (() => Promise.resolve());
+
+    try {
+      await beforeClose();
+    } catch (_error) {
```

Comment:
Looks like `_error` here is unused.

Reply:
Yep, thats correct. Its underscored because its unused but a required param.